### ISSUE TITLE
The app@sharepoint principal is not resolving in newly created tenants

### ISSUE THREAD

**Issue by michael-jensen** [SharePoint/sp-dev-docs#6155]
## Category

- [x] Bug

## Describe the bug
In the [elevated privileges page](https://github.com/SharePoint/sp-dev-docs/blob/e49588c642eac9aff573ba84ffde2ca35ee07546/docs/solution-guidance/elevated-privileges-in-sharepoint-add-ins.md) there is an important tip about adding the app@sharepoint user as a term store administrator if you need app-only write access to the term store (I believe @wobba originally wrote about this in a [post](https://www.techmikael.com/2018/08/modifying-terms-using-app-only-tokens.html) a couple of years ago).

Unfortunately we were not able to add the app@sharepoint user to the term store administrators group in a couple of tenants that we created in the past couple of days - that account will not resolve in the old and new term store UI (as shown in the following screenshots)

Old page
![old-termstore-ui](https://user-images.githubusercontent.com/8461914/89699441-cacb9b00-d8db-11ea-83bb-c539f82b4262.png)

New page
![new-termstore-ui](https://user-images.githubusercontent.com/8461914/89699444-cdc68b80-d8db-11ea-99dc-b3f45d799599.png)

It appears this issue is not isolated to the term store, as that user would not resolve in other user management areas (i.e. site collection admin, etc.)

What made this even more confusing was I was able to get the app@sharepoint account to resolve in one of our newly created tenants this afternoon, but that only worked via the old term store UI - the other tenant we created yesterday is still not able to resolve that account.

I also tried adding the full username i:0i.t|00000003-0000-0ff1-ce00-000000000000|app@sharepoint and experienced the same result as trying to simply add app@sharepoint.

## Steps to reproduce

1. Create a new tenant
2. Go to the term store in the SharePoint admin center (you can try this in both the old and new UI)
3. Add app@sharepoint to the term store administrators and try to resolve that account (and/or save your change)
4. See the error saying no match found in the old UI (or no error, but no user resolution in the new UI)

## Expected behavior

I expect the app@sharepoint account to resolve, so we can continue to use app only principals to write to the term store.

## Environment details (*development & target environment*)

- **Your Developer Environment**: N/A
- **Target Environment**: SharePoint Online
- **Framework**: N/A
- **Browser(s)**: Chrome v84
- **Tooling**: N/A
- **Additional details**: N/A

### Additional context

My concern is this app@sharepoint account may be in the process of being removed, which means our app only apps will no longer be able to write to the term store (which would obviously be a significant issue). @VesaJuvonen do you have any insight into what might be happening here?

*Thanks for your contribution! Sharing is caring.*

**Comment by wobba** [comment]
Curious, if you create an app catalog, would that help? Seems something might trigger the creation of this system account.

**Comment by michael-jensen** [comment]
Thanks for suggestion @wobba 

I tried that on a new tenant I created today, and the app@sharepoint account does not resolve (in both the old and new term store management pages). I'll try again at different intervals (i.e. 6, 12, 24 hours) to see if the process of creating a tenant app catalog triggers the process of that system account; I'll leave another comment with the results of my test.

**Comment by wobba** [comment]
Some info in the comments here https://sharepoint.stackexchange.com/questions/168160/what-is-sharepoint-app-principal-shown-in-appprincipals-aspx-and-what-created-it but nothing definitive on how this works.

**Comment by michael-jensen** [comment]
Thanks for additional suggestion @wobba 

I tried creating and registering an app via /_layouts/15/AppRegNew.aspx and /_layouts/15/AppInv.aspx with the following permissions

`<AppPermissionRequests AllowAppOnlyPolicy="true">
	<AppPermissionRequest Scope="http://sharepoint/taxonomy" Right="Write" />
</AppPermissionRequests>`

in the root site collection in the tenant and I am still not able to resolve the app@sharepoint account (after waiting 6 hours).

I should note that I created the tenant app catalog using the PnP cmdlet `New-PnPTenantSite -Template "APPCATALOG#0" ...` so I tried spinning up another tenant and created the tenant app catalog through the admin center UI in case there was something hidden happening behind the scenes when you create the tenant app catalog via the browser - unfortunately (or perhaps fortunately) I ended up with the same results: I cannot resolve the app@sharepoint account in that tenant either (I have waited a number of hours, and created the same app in that tenant as well).

Out of curiosity, what do you experience if you spin up a new tenant https://portal.office.com/Signup?OfferId=B07A1127-DE83-4a6d-9F85-2C104BDAE8B4&dl=ENTERPRISEPACK&ali=1 and try to get the app@sharepoint account to resolve?

**Comment by andrewconnell** [comment]
I'm not sure this is a developer topic... we're supposed to direct questions like this that are more OOTB to TechCommunity. Might be best to post this question there...

**Comment by wobba** [comment]
I'll post my follow up. I experienced the same and got it to work like this:

1. Create a new app with app-only permissions following https://docs.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azureacs
2. Connected to -admin and was able to resolve `Get-PnPUser -Identity "i:0i.t|00000003-0000-0ff1-ce00-000000000000|app@sharepoint"`
3. Added `i:0i.t|00000003-0000-0ff1-ce00-000000000000|app@sharepoint` as a user in term store

Not sure if 2 is needed
